allennlp/training/learning_rate_schedulers.py

```diff
@@ -126,6 +126,100 @@ def get_lr(self):
 
         return [scale for _ in range(len(self.base_lrs))]
 
+
+class STLR(torch.optim.lr_scheduler._LRScheduler): # pylint: disable=protected-access
```

Comment:
For tests, I would recommend having a simple unit test for each of the methods on this class, and a test that makes sure this can be instantiated correctly with `LearningRateScheduler.from_params(optimizer, some_param_object)`.